config/v1alpha2/types_insights.go

```diff
@@ -0,0 +1,115 @@
+package v1alpha2
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+//
+// InsightsDataGather provides data gather configuration options for the the Insights Operator.
+//
+// Compatibility level 4: No compatibility is provided, the API can change at any point for any reason. These capabilities should not be used by applications needing long term support.
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:path=insightsdatagathers,scope=Cluster
+// +kubebuilder:subresource:status
+// +openshift:api-approved.openshift.io=https://github.com/openshift/api/pull/1245
+// +openshift:file-pattern=cvoRunLevel=0000_10,operatorName=config-operator,operatorOrdering=01
+// +openshift:enable:FeatureGate=InsightsConfig
+// +openshift:compatibility-gen:level=4
+type InsightsDataGather struct {
+	metav1.TypeMeta `json:",inline"`
+	// metadata is the standard object's metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	// spec holds user settable values for configuration
+	// +required
+	Spec InsightsDataGatherSpec `json:"spec"`
+	// status holds observed values from the cluster. They may not be overridden.
+	// +optional
+	Status InsightsDataGatherStatus `json:"status"`
+}
+
+type InsightsDataGatherSpec struct {
+	// gatherConfig spec attribute includes all the configuration options related to
+	// gathering of the Insights data and its uploading to the ingress.
+	// +optional
+	GatherConfig GatherConfig `json:"gatherConfig,omitempty"`
+}
+
+type InsightsDataGatherStatus struct{}
+
+// gatherConfig provides data gathering configuration options.
+type GatherConfig struct {
+	// dataPolicy allows user to enable additional global obfuscation of the IP addresses and base domain
+	// in the Insights archive data. Valid values are "ClearText" and "ObfuscateNetworking".
+	// When set to ClearText the data is not obfuscated.
+	// When set to ObfuscateNetworking the IP addresses and the cluster domain name are obfuscated.
+	// When omitted, this means no opinion and the platform is left to choose a reasonable default, which is subject to change over time.
+	// The current default is ClearText.
+	// +optional
+	DataPolicy DataPolicy `json:"dataPolicy,omitempty"`
+	// gatherers is a list of gatherers configurations.
+	// The particular gatherers IDs can be found at https://github.com/openshift/insights-operator/blob/master/docs/gathered-data.md.
+	// Run the following command to get the names of last active gatherers:
+	// "oc get insightsoperators.operator.openshift.io cluster -o json | jq '.status.gatherStatus.gatherers[].name'"
+	// +kubebuilder:validation:MaxItems=100
+	// +listType=atomic
```

Comment:
I'm surprised this is atomic, aren't the gatherers all uniquely named/id'd?

Reply:
I was struggling with this because I initially had `+listType=set`, but the tests were failing since GathererConfig is not an atomic type. So, I iterated to +`listType=atomic`, which passed the tests. I'm not entirely sure what the correct approach is, but the gatherers do have unique names. Maybe we don’t need `+listType` at all?

Reply:
You want `+listType=map` with `+listMapKey=name` I think, each gatherer has a unique name?

Reply:
Yes, gatherers have unique names